src/classes/STG_InstallScript.cls

```diff
@@ -337,25 +338,72 @@ global without sharing class STG_InstallScript implements InstallHandler {
             UTIL_OrgTelemetry.dispatchTelemetryBatchJob();
         }
     }
-    
-    /**********************************************************************************************************
-    * @description Enables prevent deletion by default on Hierarchy Custom Settings for new installs and 
-    * when upgrading to 1.95
+
+    /**************************************************************************************************************
+    * @description Enables Prevent Hierarchy Custom Settings for new installs and push upgrades
     * @return void
     */
     global static void enablePreventDeletionHierarchyCustomSettings(Version previousVersion) {
-        if ((previousVersion == NULL) || (previousVersion != null && previousVersion.compareTo(new Version(1,95)) < 0)) {
+        Hierarchy_Settings__c orgSettings = UTIL_CustomSettingsFacade.getSettings();
+        Boolean isNewInstallOrPushUpgradedTo195 = isNewInstallOrPushUpgrade(previousVersion, new Version(1,95));
+        Boolean isNewInstallOrPushUpgradedTo196 = isNewInstallOrPushUpgrade(previousVersion, new Version(1,96));
```

Comment:
@MatthewBlanski if there are any Customers still on 1.94 and when they upgrade to 1.95 'isNewInstallOrPushUpgrade(previousVersion, new Version(1,96)' would return true because 1.94 < 1.96 as a result 'Prevent_Program_Plan_Deletion__c' is marked as true along with 'Prevent_Plan_Requirement_Deletion__c' and 'Prevent_Account_Deletion__c'. Not sure if it is a valid scenario and if so it is ok to make 'Prevent_Plan_Requirement_Deletion__c' to true in that case so wanted to check with you. TIA.

Reply:
@skosgi1 Since the intent behind this is to handle upgrade scenarios without being disruptive to a production org, this would be desired behavior as we are effectively defaulting these prevent deletion settings to true. Theoretically, if someone were capable of upgrading from version 1.0 all the way to the current version, we would want the entire run to process to bring the org up to speed.